packages/pwa-kit-react-sdk/src/utils/performance.js

```diff
@@ -78,50 +77,148 @@ export default class PerformanceTimer {
      * This is a utility function to create performance marks.
      * The data will be used in console logs and the http response header `server-timing`.
      *
-     * @function
-     * @private
+     * @param {string} name - Unique identifier for the performance measurement.
+     * Must be the same for both start and end marks of a pair. E.g. 'ssr.render-to-string'
+     *
+     * @param {string} type - Mark type, either 'start' or 'end'. 'start' creates spans and browser marks,
+     * 'end' completes measurement and cleanup.
+     *
+     * @param {Object} [options={}] - Optional configuration object
+     * @param {string|Object} [options.detail=''] - Additional metadata for the mark
+     * included in logs and tracing attributes.
      */
     mark(name, type, options = {}) {
-        if (!this.enabled) {
+        const {detail = ''} = options
+        if (!name || !type || !this.enabled) {
```

Comment:
Should we keep checking if enabled first, then validate inputs?

```suggestion
        if (!this.enabled || !name || !type) {
```